src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -137,21 +137,59 @@ final case class EServicesApiServiceImpl(
     }
   }
 
-  override def createDescriptor(eServiceId: String, eServiceDescriptorSeed: EServiceDescriptorSeed)(implicit
+  override def createDescriptor(eServiceId: String)(implicit
     contexts: Seq[(String, String)],
     toEntityMarshallerProblem: ToEntityMarshaller[Problem],
     toEntityMarshallerCreatedResource: ToEntityMarshaller[CreatedResource]
   ): Route = {
+
+    def cloneDocument(
+      clonedDocumentId: UUID,
+      doc: CatalogProcess.EServiceDoc
+    ): Future[CatalogProcess.CreateEServiceDescriptorDocumentSeed] =
+      fileManager
+        .copy(ApplicationConfiguration.eServiceDocumentsContainer, ApplicationConfiguration.eServiceDocumentsPath)(
+          doc.path,
+          clonedDocumentId.toString,
+          doc.name
+        )
+        .map(clonedPath =>
+          CatalogProcess.CreateEServiceDescriptorDocumentSeed(
+            documentId = clonedDocumentId,
+            kind = CatalogProcess.EServiceDocumentKind.DOCUMENT,
+            contentType = doc.contentType,
+            prettyName = doc.prettyName,
+            fileName = doc.name,
+            filePath = clonedPath,
+            checksum = doc.checksum,
+            serverUrls = Seq()
+          )
+        )
+
     val result: Future[CreatedResource] = for {
-      eServiceUuid <- eServiceId.toFutureUUID
-      descriptor   <- catalogProcessService.createDescriptor(eServiceUuid, eServiceDescriptorSeed.toProcess)(contexts)
+      eServiceUuid       <- eServiceId.toFutureUUID
+      eService           <- catalogProcessService.getEServiceById(eServiceUuid)
+      previousDescriptor <- Either
+        .cond(eService.descriptors.nonEmpty, eService.descriptors.maxBy(_.version), NoDescriptorsFound(eServiceUuid))
+        .toFuture
+      clonedDocuments    <- Future.traverse(previousDescriptor.docs)(cloneDocument(UUIDSupplier.get(), _))
+      eServiceDescriptorSeed = CatalogProcess.EServiceDescriptorSeed(
+        description = previousDescriptor.description,
+        audience = Seq(),
```

Comment:
won't this fail for [constraint](https://github.com/pagopa/interop-be-monorepo/pull/739/files#diff-09c9a8d490c6f40553523dd2c43b8e1eea1de7e649804d4ba204466f46df2696L1151) on process?

Reply:
Interesting! So, since we don't have a default for that, we should decrease the minimum length to 0.

Reply:
it could be an option, but notice that in that case we need to validate the audience when publishing the descriptor (now it's not required because it is enforced by the spec).

I suppose the audience should not be copied from the previous descriptor?

Reply:
Update: the constraint is being edited (https://github.com/pagopa/interop-be-monorepo/pull/762)